File: webapp/airTicket/src/app/core/service/city.service.ts
import {Injectable} from '@angular/core';
import {CoreModule} from '../core.module';
import {Observable, of} from 'rxjs';
import {City} from '../../entity/City';
import {HttpClient} from '@angular/common/http';

@Injectable({
  providedIn: CoreModule
})
export class CityService {

  baseUrl = 'api/city';

  constructor(private httpClient: HttpClient) {
  }

  // 保存城市
  save(city: City): Observable<City> {
    return this.httpClient.post<City>(this.baseUrl, city);
  }

  // 通过id获取城市
  getById(id: number): Observable<City> {
    return this.httpClient.get<City>(this.baseUrl + '/' + id);
  }

  // 更新
  update(id: number, city: City): Observable<City> {
    return this.httpClient.put<City>(this.baseUrl + '/' + id, city);
  }

  // 删除
  delete(id: number): Observable<any> {
    return this.httpClient.delete(this.baseUrl + '/' + id);
  }

  // 获取所有城市
  getAll(): Observable<City[]> {
    return this.httpClient.get<City[]>(this.baseUrl);
  }

  // 设为热门城市
  setToPrimaried(id: number): Observable<any> {
    return this.httpClient.put(this.baseUrl + '/toPrimaried/' + id, {});
  }

  // 取消热门城市
  setToNotPrimaried(id: number): Observable<any> {
    return this.httpClient.put(this.baseUrl + '/notToPrimaried/' + id, {});
  }

  // 全部设为热门城市
  setAllToPrimaried(ids: Array<number>): Observable<any> {
    return this.httpClient.post(this.baseUrl + '/allToPrimaried', ids);
  }

  // 全部取消设为热门城市
  setAllToNotPrimaried(ids: Array<number>): Observable<any> {
    return this.httpClient.post(this.baseUrl + '/allNotToPrimaried', ids);
  }

}

File: api/airTicket/src/main/java/com/xiang/airTicket/service/CityService.java
package com.xiang.airTicket.service;

import com.xiang.airTicket.entity.City;

import java.util.List;

public interface CityService {
    // 保存全部城市
    List<City> saveAll(Iterable<City> cities);

    // 获取全部城市
    Iterable<City> getAll();

    // 获取指定城市
    City getById(Long id);

    // 保存城市
    City save(City city);

    // 更新城市
    City update(Long id, City city);

    // 删除城市
    void delete(Long id);

    // 设为主要城市
    void setToPrimaried(Long id);

    void setToNotPrimaried(Long id);

    // 全部设为主要城市
    void setAllToPrimaried(List<Long> ids);

    // 全部取消设为主要城市
    void setAllToNotPrimaried(List<Long> ids);
}
